### PR TITLE
popup-menu

### PR DESCRIPTION
**Pull request by Eugene50**
### HTML-CSS-Popup

[Demo](https://eugene50.github.io/popup-menu/) |
[Code base](https://github.com/Eugene50/popup-menu)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by Eugene50** [comment]
Please checkout)!

File: submissions/Eugene50/popup-menu/index.html [hunk 1]
```diff
@@ -33,11 +33,11 @@
...
 
             <input id="sub-checkbox" type="checkbox">
-            <label id="more-label" for="sub-checkbox">
+            <label class="more-label" for="sub-checkbox">
                     <p>More</p>
             </label>
             
-            <div id="sub-container">
+            <div class="sub-container">
                 <a href="#"><div class="item" tabindex="11"><img src="img/1.png" alt="">Advertising</div></a>
```
Comment (A-Ostrovnyy) [inline submissions/Eugene50/popup-menu/index.html]: Fill `alt` attributes.

File: submissions/Eugene50/popup-menu/style.css [hunk 2]
```diff
@@ -118,27 +118,27 @@ img {
...
  }
 
-#alf {
+.alf {
    border-radius: 180px;
 }
 
 #main-checkbox {
-   visibility: hidden;
+   opacity: 0;
```
Comment (A-Ostrovnyy) [inline submissions/Eugene50/popup-menu/style.css]: [How to correct to hide checkbox.](https://youtu.be/E6kLaaQFctU)

**Comment by Eugene50** [comment]
update code, please chekout!

**Comment by A-Ostrovnyy** [comment]
@Eugene50 please watch [video](https://www.youtube.com/watch?v=E6kLaaQFctU&feature=youtu.be) how to hide checkbox and apply this method to your popup.

**Comment by Eugene50** [comment]
I am forgot to update code base, i am apply method from video! please check)
P.s. i`m delete oppacity and don`t use visability: hidden.

File: submissions/Eugene50/popup-menu/style.css [hunk 3]
```diff
@@ -53,6 +53,14 @@ img {
    margin-left: 0px;
 }
 
+#main-checkbox {
```
Comment (A-Ostrovnyy) [inline submissions/Eugene50/popup-menu/style.css]: Avoid styling by `#id`.
Separate responsibilities: HTML tag - marckup, Clasess - styles, #id - javaScritp(logic).

**Comment by Eugene50** [comment]
I hope I`m done) please check!

File: submissions/Eugene50/popup-menu/index.html [hunk 4]
```diff
@@ -11,7 +11,7 @@
     <div class="container">
         <div class="menu-item"><a href="#">Mail</a></div>
         <div class="menu-item"><a href="#">Blog</a></div>
-        <input id="main-checkbox" type="checkbox">
+        <input class="main-checkbox" type="checkbox">
```
Comment (A-Ostrovnyy) [inline submissions/Eugene50/popup-menu/index.html]: You need `#main-checkbox` for linking `label` with input and also,  you need `.main-checkbox` for adding styles to input.

File: submissions/Eugene50/popup-menu/style.css [hunk 5]
```diff
@@ -53,45 +53,45 @@ img {
    margin-left: 0px;
 }
 
-#main-checkbox {
+.main-checkbox {
```
Comment (A-Ostrovnyy) [inline submissions/Eugene50/popup-menu/style.css]: You forget to add `overflow: hidden;` property.
Also, you have `.sub-checkbox` and `.main-checkbox` classes with equal properties, maybe better use one abstract class,  `.visually-hidden` as an example, for hiding default inputs control.
**D**on't
**R**epeat
**Y**ourself

File: submissions/Eugene50/popup-menu/index.html [hunk 6]
```diff
@@ -11,7 +11,7 @@
...
         <label class="menu-label" for="main-checkbox">
             <img class="menu-icon" src="img\menu.png" height="20" width="20"/>
```
Comment (A-Ostrovnyy) [inline submissions/Eugene50/popup-menu/index.html]: Don't forget fill `alt` attribute on the `` tag.

File: submissions/Eugene50/popup-menu/index.html [hunk 7]
```diff
@@ -8,18 +8,18 @@
...
+        <div class="blog" class="menu-item"><a href="#">Blog</a></div>
         <input id="main-checkbox" type="checkbox">
-        <label id="menu-label" for="main-checkbox">
-            <img id="menu-icon" src="img\menu.png" height="20" width="20"/>
+        <label class="menu-label" for="main-checkbox">
+            <img class="menu-icon" src="img\menu.png" height="20" width="20"/>
         </label>
         <div class="menu-item"><a href="#">Music</a></div>
-        <div><a href="#"><img id="alf" src="img/alf.jpg" alt="User"></a></div>
+        <div><a href="#"><img class="alf" src="img/alf.jpg" alt="User"></a></div>
```
Comment (A-Ostrovnyy) [inline submissions/Eugene50/popup-menu/index.html]: Avoid using `<div>` without class as a wrapper.

**Comment by Eugene50** [comment]
please check)!!

File: submissions/Eugene50/popup-menu/index.html [hunk 8]
```diff
@@ -11,12 +11,12 @@
...
         <div class="menu-item"><a href="#">Blog</a></div>
-        <input class="main-checkbox" type="checkbox">
+        <input id="main-checkbox" class="main-checkbox visually-hidden" type="checkbox">
         <label class="menu-label" for="main-checkbox">
-            <img class="menu-icon" src="img\menu.png" height="20" width="20"/>
+            <img class="menu-icon" src="img\menu.png" alt="Menu" height="20" width="20"/>
         </label>
         <div class="menu-item"><a href="#">Music</a></div>
-        <div><a href="#"><img class="alf" src="img/alf.jpg" alt="User"></a></div>
+        <div class="menu-icon-alf"><a href="#"><img class="alf" src="img/alf.jpg" alt="User"></a></div>
```
Comment (A-Ostrovnyy) [inline submissions/Eugene50/popup-menu/index.html]: Class `.menu-icon-alf` without styles, why you use him?

**Comment by Eugene50** [comment]
Update code, please check!

**Comment by A-Ostrovnyy** [comment]
Well done!

**Pull request merged**